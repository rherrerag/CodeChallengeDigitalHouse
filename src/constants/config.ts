const config = {
  FILTER: {
    MONTH: '3',
    YEAR: '2023',
  },
};

export default config;
