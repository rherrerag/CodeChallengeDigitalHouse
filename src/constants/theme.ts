import {extendTheme} from 'native-base';

const customTheme = extendTheme({
  colors: {
    primary: {
      100: '#334FFA',
      200: '#CFD6FF',
    },
    text: {
      title: '#020202',
      subtitle: '#9B9898',
      normal: '#000000',
    },
  },
});

export {customTheme};
