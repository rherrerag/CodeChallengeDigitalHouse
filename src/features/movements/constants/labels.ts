const labels = {
  WELCOME: {
    MESSAGE: 'Bienvenido de Vuelta!',
    NAME: 'Ruben Rodriguez',
  },
  SUBTITLES: {
    POINTS: 'TUS PUNTOS',
    MOVEMENTS: 'TUS MOVIMIENTOS',
  },
  CARD: {
    POINTS: 'pts',
  },
  BUTTONS: {
    EARNED: 'Ganados',
    REDEEMED: 'Canjeados',
    ALL: 'Todos',
  },
  LOADING: {
    TITLE: 'Cargando',
  },
};

export default labels;
