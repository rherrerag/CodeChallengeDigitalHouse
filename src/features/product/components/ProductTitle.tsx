import React from 'react';
import {ProductTitleContainer, ProductTitleText} from './ProductTitle.styled';

type ProductTitleProps = {
  title?: string;
};

const ProductTitle: React.FC<ProductTitleProps> = ({title}) => {
  return (
    <ProductTitleContainer bg="primary.200">
      <ProductTitleText>{title}</ProductTitleText>
    </ProductTitleContainer>
  );
};
export default ProductTitle;
