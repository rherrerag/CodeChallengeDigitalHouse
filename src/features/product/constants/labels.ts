const labels = {
  PRODUCT_DETAIL: {
    SUBTITLE_DETAIL: 'Detalles del producto:',
    SUBTITLE_SUMMARY: 'Con esta compra acumulaste:',
    POINTS_TEXT: 'puntos',
    DATE_SUBSTRING: 'Comprado el',
  },
  ACCEPT_BUTTON: {
    TITLE: 'Aceptar',
  },
};

export default labels;
