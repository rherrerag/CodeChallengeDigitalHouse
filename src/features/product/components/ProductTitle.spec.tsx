import React from 'react';
import {render, screen} from '@testing-library/react-native';
import ProductTitle from './ProductTitle';
import {NativeBaseProvider} from 'native-base';
import {customTheme} from '../../../constants/theme';

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

describe('ProductTitle Component', () => {
  it('deberÃ­a generar snapshot de componente correctamente', async () => {
    render(
      <NativeBaseProvider theme={customTheme} initialWindowMetrics={inset}>
        <ProductTitle title="Titulo de Prueba" />
      </NativeBaseProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
