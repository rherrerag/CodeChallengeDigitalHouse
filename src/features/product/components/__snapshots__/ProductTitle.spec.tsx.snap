// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Component deber√≠a generar snapshot de componente correctamente 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      [
        {
          "backgroundColor": "#CFD6FF",
        },
        [
          {
            "paddingBottom": 24,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 80,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      dataSet={{}}
      style={
        [
          {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          },
          [
            {
              "fontSize": 24,
              "fontWeight": "800",
              "lineHeight": 24,
            },
            undefined,
          ],
        ]
      }
    >
      Titulo de Prueba
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
